### PR TITLE
Added the initial gradle build with scala support.

### PR DESCRIPTION
**Pull request by meln1k**
Summary: now we have a build structure with scala application support.

### CODE REVIEW

**Comment by jeschkies** [comment]
Should we change the `.gitignore` for Gradle files?

File: src/main/scala/com/mesosphere/usi/Main.scala [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package com.mesosphere.usi
```
Comment (jeschkies) [inline src/main/scala/com/mesosphere/usi/Main.scala]: I think we should start right away with a multi-project setup. What do you think about
* `interface`
* `core`
* `hello-world`

With `core` depending on `interface` and `hello-world` depending on both and being a runnable?

File: src/main/scala/com/mesosphere/usi/Main.scala [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+
+object Main extends App {
+
+  println("Starting orchestrator")
```
Comment (zen-dog) [inline src/main/scala/com/mesosphere/usi/Main.scala]: Don't think we're going to "start an orchestrator" here ;)

File: src/test/scala/com/mesosphere/usi/DummySuite.scala [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+ */
+package com.mesosphere.usi
+
+import org.junit.runner.RunWith
+import org.scalatest.FunSuite
+import org.scalatest.junit.JUnitRunner
+
+@RunWith(classOf[JUnitRunner])
+class DummySuite extends FunSuite {
+  test("true is always true") {
```
Comment (zen-dog) [inline src/test/scala/com/mesosphere/usi/DummySuite.scala]: Isn't it a philosophical question? :D

**Review by zen-dog — APPROVED** [review]
🚢

**Review by jeschkies — CHANGES_REQUESTED** [review]

File: src/main/scala/com/mesosphere/usi/Main.scala (hunk 1 above)
Comment (meln1k) [reply src/main/scala/com/mesosphere/usi/Main.scala]: Sounds good, will do in a second

File: build.gradle [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+ * This file was generated by the Gradle 'init' task.
+ *
+ * This generated file contains a sample Scala library project to get you started.
+ * For more details take a look at the Scala plugin chapter in the Gradle
+ * user guide available at https://docs.gradle.org/5.1.1/userguide/scala_plugin.html
+ */
+
+plugins {
+    // Apply the scala plugin to add support for Scala
+    id 'scala'
```
Comment (jeschkies) [inline build.gradle]: Why is this in `build.gradle` and not `example/build.gradle`?

File: build.gradle [hunk 5]
```diff
@@ -0,0 +1,52 @@
+/*
+ * This file was generated by the Gradle 'init' task.
+ *
+ * This generated file contains a sample Scala library project to get you started.
+ * For more details take a look at the Scala plugin chapter in the Gradle
+ * user guide available at https://docs.gradle.org/5.1.1/userguide/scala_plugin.html
```
Comment (jeschkies) [inline build.gradle]: I don't think we need these comments.

File: core/build.gradle [hunk 6]
```diff
@@ -0,0 +1,8 @@
+plugins {
+    // Apply the scala plugin to add support for Scala
+    id 'scala'
+}
+
+dependencies {
+    compile project(':interface')
```
Comment (jeschkies) [inline core/build.gradle:2]: Nice!

File: example-orchestrator/build.gradle [hunk 7]
```diff
@@ -0,0 +1,15 @@
+plugins {
```
Comment (jeschkies) [inline example-orchestrator/build.gradle]: We called the example `hello-world` if I'm not mistaken.

File: settings.gradle [hunk 8]
```diff
@@ -0,0 +1,14 @@
+/*
+ * This file was generated by the Gradle 'init' task.
+ *
+ * The settings file is used to specify which projects to include in your build.
+ *
+ * Detailed information about configuring a multi-project build in Gradle can be found
+ * in the user guide at https://docs.gradle.org/5.1.1/userguide/multi_project_builds.html
```
Comment (jeschkies) [inline settings.gradle]: I don't think these are required.

File: interface/src/main/scala/com/mesosphere/usi/interface/DummySchedulerInterface.scala [hunk 9]
```diff
@@ -0,0 +1,20 @@
+package com.mesosphere.usi.interface
+
+import com.mesosphere.usi.interface.DummySchedulerInterface._
```
Comment (jeschkies) [inline interface/src/main/scala/com/mesosphere/usi/interface/DummySchedulerInterface.scala]: No wildcard imports 😉

**Review by jeschkies — CHANGES_REQUESTED** [review]
Thanks. This looks much better to me. The `build.gradle` in root should be much leaner, though. At least it feels like that to me.

**Review by jeschkies — APPROVED** [review]
Given you discuss the example name with @meichstedt this looks great to me!

**Pull request merged**